packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx

```diff
@@ -1,383 +1,60 @@
-          enabled: true,
-          phase: 'afterWrite' as PopperJs.ModifierPhases,
-          fn: handleUpdate,
-        },
-        popperInitialPositionFix.modifier,
-
-        autoSize && {
-          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
-          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
-          name: 'applyMaxSize',
-          enabled: true,
-          phase: 'beforeWrite' as PopperJs.ModifierPhases,
-          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
-          options: {
-            altBoundary: true,
-            boundary: getBoundary(contentRef.current, overflowBoundary),
-          },
-          fn({ state, options }: PopperJs.ModifierArguments<{}>) {
-            const overflow = PopperJs.detectOverflow(state, options);
-            const { x, y } = state.modifiersData.preventOverflow || { x: 0, y: 0 };
-            const { width, height } = state.rects.popper;
-            const [basePlacement] = state.placement.split('-');
-
-            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
-            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
-
-            if (overflow[widthProp] > 0 && (autoSize === true || autoSize === 'width')) {
-              state.styles.popper.maxWidth = `${width - overflow[widthProp] - x}px`;
-            }
-            if (overflow[heightProp] > 0 && (autoSize === true || autoSize === 'height')) {
-              state.styles.popper.maxHeight = `${height - overflow[heightProp] - y}px`;
-            }
-          },
-        },
-      ].filter(Boolean),
-      onFirstUpdate: state => handleUpdate({ state }),
-    };
-
-    popperInstanceRef.current = PopperJs.createPopper(reference, contentRef.current, options);
-    popperInitialPositionFix.patch(popperInstanceRef.current);
-  }, [
-    contentRef,
-    computedModifiers,
-    enabled,
-    flipBoundary,
-    overflowBoundary,
-    pointerTargetRef,
-    positionFixed,
-    proposedPlacement,
-    targetRef,
-    unstable_disableTether,
-    unstable_pinned,
-    popperInitialPositionFix,
-    autoSize,
-  ]);
-
-  const destroyInstance = React.useCallback(() => {
-    if (popperInstanceRef.current) {
-      popperInstanceRef.current.destroy();
-      popperInstanceRef.current = null;
-    }
-  }, []);
-
-  const scheduleUpdate = React.useCallback(() => {
-    if (popperInstanceRef.current) {
-      popperInstanceRef.current.update();
-    }
-  }, []);
-
-  React.useImperativeHandle(
-    popperRef,
-    () => ({
-      updatePosition: scheduleUpdate,
-    }),
-    [scheduleUpdate],
-  );
+        // A state change has sense only if renderProps are passed, otherwise a state value is unused
+        if (usesRenderProps) {
+          setComputedPlacement(state.placement);
+        }
```

Comment:
Just really really curious why this set state should only be called for render props? couldn't figure it out myself :(

Reply:
_What is `computedPlacement`?_ It's an actual position of a container after positioning of flipping and it can be different than `proposedPlacement` (user' input). We have a following condition in `Popper.tsx`:

```tsx
  const child = usesRenderProps
    ? (props.children as PopperChildrenFn)({
        placement: computedPlacement, // ⚠️ computedPlacement is used only there
        scheduleUpdate: popperRef.current?.updatePosition,
      })
    : (props.children as React.ReactElement);
```

We will use `computedPlacement` only when a component will use render props:

```tsx
// when a function is passed, we will pass `computedPlacement` down
<Popper>{({ placement }) => <div className={placement} />}</Popper>
// when an element passed, we don't use `computedPlacement`
<Popper><div /></Popper>
```

Reply:
Very nice explanation!